### ISSUE TITLE
[2nd iteration] Soft deprecation for ".wazuh" index

### ISSUE THREAD

**Issue by jesusgn90** [wazuh/wazuh-kibana-app#1771]
Hi team,

After #1465 and after discussing internally how we should manage this task, here are the next steps for this task.

#### Client side (front-end) 

- [x] If there are no API entries, show a guide. @jesusgn90 
  - [x] Explain how to add an API entry.
  - [x] Add an option for testing the connection in the same guide.
- [x] If there are API entries, but we can't reach the API, show a guide.  @jesusgn90 
  - [x] This guide shows how to check the API service, how to configure an API entry.
  - [x] This guide also shows the existing API entries plus a button for checking the connection of all of them.
- [x] Remove the form in Settings @jesusgn90 
  - [x] Also change the API entries table (no actions except "test connection" and "set as default")

#### Server side (back-end)

- [x] If the index `.wazuh` exists, copy its content to `wazuh-hosts.yml` @adri9valle 
  - [x] Decode base64 passwords from API entries.
  - [x] Ensure all API entries were copied properly before removing the index.
  - [x] Remove the index once done.
  - [x] Do not duplicate entries.

- [x] Every API entry found in `wazuh-hosts.yml` has a pair in `wazuh-registry.json` (old `wazuh-version.json`) with the cluster status, the manager name, and the related extensions. @adri9valle 
  - [x] Adapt our server endpoints for updating API properties, the front-end should be able to use the same endpoints as before, not changing its logic drastically.
  - [x] Handle race conditions (two persons using the app can generate this easily).
- [x] Remove orphan code after the rework, for example, the "Save settings" option in the front-end is going to be removed, then the related back-end route is no longer needed.  
- [x] Remove orphan registry entries when a host is deleted.

Default structure for `wazuh-hosts.yml`:

```yml
# Wazuh API entries
wazuh.hosts:
#  - <id>:
#     url: <url>
#     port: <port>
#     user: <user>
#     password: <password> 
```
Real example:
```yml
# Wazuh API entries
wazuh.hosts:
  - production:
     url: https://10.0.0.2
     port: 55000
     user: admin
     password: kj178hkjasdf2
  - staging:
     url: http://localhost
     port: 55000
     user: foo
     password: bar
```
In the above example, both `production` and `staging` are API entries. Each one would have an entry in the `wazuh-registry.json` with the extensions and the cluster information.

**Comment by jesusgn90** [comment]
Hi team, I've been working on the front-end side of this task. The working branch is https://github.com/wazuh/wazuh-kibana-app/tree/issue-1771-1.

Started two new components for "How to configure the Wazuh API" and "Troubleshooting Wazuh API not reachable", not finished yet, but here is a sneak-peek:

![image](https://user-images.githubusercontent.com/8860227/64357919-d0eb3480-d005-11e9-9a2f-543c81d9eddb.png)

![image](https://user-images.githubusercontent.com/8860227/64358000-f11af380-d005-11e9-9d56-5888d62d3a53.png)

![image](https://user-images.githubusercontent.com/8860227/64357945-dcd6f680-d005-11e9-88b2-cf8b0a0d4cb5.png)

![image](https://user-images.githubusercontent.com/8860227/64357960-e3656e00-d005-11e9-8351-f1c7976d41f9.png)

On the other hand, I'm also working on changing the "Settings" controller, some functions are no longer needed, other ones are wrong now and must be adapted for this task.

**Comment by adri9valle** [comment]
**Update 09.06.19**

The following points related to this issue are solved in this branch: [migrate-index-yml](https://github.com/wazuh/wazuh-kibana-app/tree/migrate-index-yml).

- If the index .wazuh exists, copy its content to wazuh-hosts.yml
  - Decode base64 passwords from API entries.
  - Ensure all API entries were copied properly before removing the index.
  - Remove the index once done.
  - Do not duplicate entries.

**Comment by jesusgn90** [comment]
After the view changes on the client-side, we've refactored the controller https://github.com/wazuh/wazuh-kibana-app/commit/dcee4c137a6feaedc87a0158c968a83164c55603 to remove unused methods, and to adapt the controller for latest changes.

**Comment by adri9valle** [comment]
**Update 09.09.19**

The `wazuh-registry.json` is updated with the cluster information and extensions of the migrated host once the host is migrated from the index to the registry file.

- Every API entry found in wazuh-hosts.yml has a pair in `wazuh-registry.json` (`old wazuh-version.json`) with the cluster status, the manager name, and the related extensions.

**Comment by adri9valle** [comment]
**Updated 10.09.19**

- The API entries in the `Settings > API` section are loaded from the `wazuh-hosts.yml`.
- When checking connection the registry is updated instead of update the index.

**Comment by adri9valle** [comment]
**Updated 11.09.19**

**Working on**:  Adapt our server endpoints for updating API properties, the front-end should be able to use the same endpoints as before, not changing its logic drastically.

Some endpoints have been changed:

- Updated the selected API.
- Check manager connection.
- Update cluster information.

**Comment by adri9valle** [comment]
**Update 13.09.19**

**Working on**: Adapt our server endpoints for updating API properties, the front-end should be able to use the same endpoints as before, not changing its logic drastically.

- Removed `wazuh-api-elastic` backend endpoints and controllers.
- Cleaned the initialize in order to remove unused and deprecated functions.

**Comment by adri9valle** [comment]
**Update 16.09.19**

- Modified the api-table component.
- Some fixes in the settings-wizard.
- Fix when joining the hosts with the registry entries.
- Add API component first poc:

Cannot reach any API: 

![any-api](https://user-images.githubusercontent.com/7983984/64970021-47f5b800-d8a5-11e9-9492-4648cc031d44.gif)

Some API can be reached others cannot:
![some-errors](https://user-images.githubusercontent.com/7983984/64970051-56dc6a80-d8a5-11e9-8e97-328f636d7214.gif)

All API entries are reached:
![ok-api](https://user-images.githubusercontent.com/7983984/64970083-6b206780-d8a5-11e9-8f0d-94379f75ab79.gif)

**Comment by adri9valle** [comment]
**Update 17.09.19**

- Improve checks for new APIs
- Test if API is down in wazuh-api
- Check down APIs improved(now when test the API connectivity you can know if is not reachable or is down).
- Api is down component finish(**can be changed**):

![api-is-down](https://user-images.githubusercontent.com/7983984/65054032-3677e300-d96d-11e9-9e71-db43bc94d729.gif)

**Comment by adri9valle** [comment]
**Update 18.09.19**

- Merged with 3.11-7.3
- Prevent create wazuh-registry.json without hosts
- Improve check manager, now it sets the API status(online, offline, unknown)
- Check APIs status when getting them
- Change  the way to display the helpers components
- Improve api-is-down component
- Adapt settings-wizard


**Current status**:
![image](https://user-images.githubusercontent.com/7983984/65159890-19a6e280-da35-11e9-880e-87d3d08deb6f.png)

**Comment by adri9valle** [comment]
**Update 19.09.19**

Improved several functionalities, also, add the refresh of the entries:

![refresh](https://user-images.githubusercontent.com/7983984/65260193-470f9100-db06-11e9-86eb-8deed6b3eb17.gif)

**Comment by adri9valle** [comment]
**Update 20.09.19**

Fixed `api-count.js`. The object structure for the API changed and it was trying to get undefined fields, this was provoking that the first time that navigates to another section different to Settings the App redirects to Settings again.

**Comment by adri9valle** [comment]
**Update 30.09.19**

- The api-is-down component only was checking the initial API entries, now on each check gets again the API entries in the wazuh.yml in order to check there is some new API entry or some entry was removed.
- Settings-wizard and api-count only tried to set as default the first entry, if this entry cannot be selected as default the APP was redirecting to `Settings`, now both try to select as default the first entry that can be recheabled.
- Fixed several typos and flicks.

**Comment by adri9valle** [comment]
**Update 01.10.19**

The error handler has been improved, until now when getting some error in the YML parser the component only was showing that an error occurred but not the reason why now this error is shown:

![image](https://user-images.githubusercontent.com/7983984/65965812-8079c200-e45f-11e9-861a-9acf5943c2fe.png)

![image](https://user-images.githubusercontent.com/7983984/65966281-275e5e00-e460-11e9-8561-7ae98a182fa7.png)

**Comment by adri9valle** [comment]
**Update 02.10.19**

Now **https** is supported.

**Comment by adri9valle** [comment]
**Update 04.10.19**

Fixed:

- When there's not a default API selected in the cookies the health-check when starting the APP thrown an error.
- When any API entry could be selected the APP was allowing the navigation, now always redirect to settings in order to manage the API entries and solve the problem.
- Warning when the select API changes.

**NOTE**:

If there's some API entry as default in the cookies when accessing the APP if the credentials have changed or the entry has been removed an error is shown in the health-check, this is normal behavior because the stored API cannot be connected, also, is the current behavior.

**Comment by adri9valle** [comment]
**Update 07.10.19**

Improved error handler when stored API cannot be reached.

**Comment by adri9valle** [comment]
**Update 08.10.19**

- Added the ID to the tables.
- Added the capability to search in the table.
- Added the capability to sort in the table.

![image](https://user-images.githubusercontent.com/7983984/66373411-e22eb480-e9a8-11e9-8d30-4e5788951fb6.png)

![image](https://user-images.githubusercontent.com/7983984/66373511-13a78000-e9a9-11e9-9afb-6bcc3bca963d.png)

**Comment by adri9valle** [comment]
**Update 09.10.19**

- Added spinner also in api-table instead the message refreshing API entries.
- Show a tooltip with the reason why the API cannot be reachable.

![image](https://user-images.githubusercontent.com/7983984/66468122-ed0b4700-ea85-11e9-8ebf-21d3a65844f6.png)

**Comment by adri9valle** [comment]
Solved: https://github.com/wazuh/wazuh-kibana-app/pull/1811
